### PR TITLE
NumericValidationBehavior: Added Max/Min decimal places

### PR DESCRIPTION
**Pull request by AndreiMisiukevich**
### Description of Change ###
Make it possible to determine allowed decimal places quantity.

Min=2, Max = 2 (**1.20** - Valid, **1.2** - Invalid)
Min=0, Max = 0 (**24** - Valid, **24.0** - Invalid, **24.** - Invalid, **.24** - Invalid)

### Bugs Fixed ###
- Related to issue #203

### API Changes ###
Added: 
 
- `int NumericValidationBehavior.MinimumDecimalPlaces { get; set; }`
- `int NumericValidationBehavior.MaximumDecimalPlaces { get; set; }`

### Behavioral Changes ###
None

### PR Checklist ###

- [X] Has tests (if omitted, state reason in description)
- [X] Has samples (if omitted, state reason in description)
- [X] Rebased on top of main at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

File: XamarinCommunityToolkit/Behaviors/Validators/NumericValidationBehavior.shared.cs [hunk 1]
```diff
@@ -22,9 +30,42 @@ public double MaximumValue
...
-			=> double.TryParse(value?.ToString(), out var numeric)
+		{
+			var valueString = value as string;
+			if (!(double.TryParse(valueString, out var numeric)
 				&& numeric >= MinimumValue
-				&& numeric <= MaximumValue;
+				&& numeric <= MaximumValue))
+				return false;
+
+			var decimalDelimetrIndex = valueString.IndexOf(decimalPartDelimeter);
```
Comment (jfversluis) [inline XamarinCommunityToolkit/Behaviors/Validators/NumericValidationBehavior.shared.cs:56]: Looks like a little typo in this variable name :D

File: XamarinCommunityToolkit/Behaviors/Validators/NumericValidationBehavior.shared.cs [hunk 2]
```diff
@@ -22,9 +30,42 @@ public double MaximumValue
...
+			// If MaximumDecimalPlaces equals zero, ".5" or "14." should be considered as invalid inputs.
+			if (hasDecimalDelimetr && MaximumDecimalPlaces == 0)
+				return false;
+
+			var decimalPlaces = hasDecimalDelimetr
+				? valueString.Substring(decimalDelimetrIndex + 1, valueString.Length - decimalDelimetrIndex - 1).Length
+				: 0;
+
+			return decimalPlaces >= MinimumDecimalPlaces
+				&& decimalPlaces <= MaximumDecimalPlaces;
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Behaviors/Validators/NumericValidationBehavior.shared.cs:68]:
```suggestion
			var decimalDelimeterIndex = valueString.IndexOf(decimalPartDelimeter);
			var hasDecimalDelimeter = decimalDelimeterIndex >= 0;

			// If MaximumDecimalPlaces equals zero, ".5" or "14." should be considered as invalid inputs.
			if (hasDecimalDelimeter && MaximumDecimalPlaces == 0)
				return false;

			var decimalPlaces = hasDecimalDelimeter
				? valueString.Substring(decimalDelimeterIndex + 1, valueString.Length - decimalDelimeterIndex - 1).Length
				: 0;

			return decimalPlaces >= MinimumDecimalPlaces
				&& decimalPlaces <= MaximumDecimalPlaces;
```

File: XamarinCommunityToolkit/Behaviors/Validators/NumericValidationBehavior.shared.cs (hunk 1 above)
Comment (AndreiMisiukevich) [reply XamarinCommunityToolkit/Behaviors/Validators/NumericValidationBehavior.shared.cs]: fixed on fly)

**Comment by AndreiMisiukevich** [comment]
@Char0394 updated (Use MaximumDecimalPlaces property in the existing sample)

**Review by jfversluis — APPROVED** [review]

**Pull request merged**